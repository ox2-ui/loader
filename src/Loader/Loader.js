import React from 'react';
import styled, { keyframes } from 'styled-components';

const Wrapper = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  display: flex;
  user-select: none;
`;

const Container = styled.div`
  align-self: center;
  margin: 0 auto;
  width: 70px;
  display: block;
  display: flex;
  padding: 20px 0;
`;

const scaleInOut = keyframes`
  0%, 80%, 100% {
    transform: scale(0.0);
  } 40% {
    transform: scale(1.0);
  }
`;

const Circle = styled.div`
  width: 18px;
  height: 18px;
  margin-left: 5px;

  border-radius: 100%;
  display: inline-block;
  animation: ${scaleInOut} 1.4s infinite ease-in-out;
  /* Prevent first frame from flickering when animation starts */
  animation-fill-mode: both;
  background-color: grey;
`;

const Circle1 = styled(Circle)`
  animation-delay: -0.32s;
`;

const Circle2 = styled(Circle)`
  animation-delay: -0.16s;
`;


/**
 * Loader Component
 */
const Loader = () => (
  <Wrapper>
    <Container>
      <Circle1 />
      <Circle2 />
      <Circle />
    </Container>
  </Wrapper>
);


export default Loader;
