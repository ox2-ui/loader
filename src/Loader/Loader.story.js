import React from 'react';
import { storiesOf } from '@kadira/storybook';
import Loader from './Loader';

storiesOf('Loader', module)
  .add('default', () => (
    <Loader />
  ));
