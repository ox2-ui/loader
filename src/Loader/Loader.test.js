/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import Loader from './Loader';

const handler = () => true;

it('renders correctly', () => {
  const wrapper = mount(
    <Loader />
  );

  expect(wrapper).toMatchSnapshot();
});
