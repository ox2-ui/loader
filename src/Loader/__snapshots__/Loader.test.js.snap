// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Loader>
  <styled.div>
    <div
      className="iqtgeR"
    >
      <styled.div>
        <div
          className="cSGQWI"
        >
          <styled.div>
            <div
              className="fcVIZN"
            />
          </styled.div>
          <styled.div>
            <div
              className="eBehfk"
            />
          </styled.div>
          <styled.div>
            <div
              className="dCaswc"
            />
          </styled.div>
        </div>
      </styled.div>
    </div>
  </styled.div>
</Loader>
`;
